src/metrics_classification.rs

```diff
@@ -493,19 +493,48 @@ impl<A: NdFloat, D: Data<Elem = A>> BinaryClassification<A> for ArrayBase<D, Ix1
 #[cfg(test)]
 mod tests {
     use super::{BinaryClassification, Classification, Modify};
+    use approx::{abs_diff_eq, AbsDiffEq};
     use ndarray::{array, Array1, ArrayBase, ArrayView1, Data, Dimension};
     use rand::{distributions::Uniform, Rng};
 
-    fn assert_eq_slice<A: std::fmt::Debug + PartialEq + Clone, S: Data<Elem = A>, D: Dimension>(
+    fn assert_eq_slice<
+        A: std::fmt::Debug + PartialEq + AbsDiffEq,
+        S: Data<Elem = A>,
+        D: Dimension,
+    >(
         a: ArrayBase<S, D>,
         b: &[A],
     ) {
-        let a = a.iter().cloned().collect::<Vec<_>>();
-        assert_eq!(a, b);
+        assert_eq_iters(a.iter(), b);
     }
 
-    fn assert_eq_iter<A: std::fmt::Debug + PartialEq>(a: impl Iterator<Item = A>, b: &[A]) {
-        assert_eq!(a.collect::<Vec<_>>(), b);
+    fn assert_eq_iter<A: std::fmt::Debug + PartialEq + AbsDiffEq + Clone>(
+        a: impl Iterator<Item = A>,
+        b: &[A],
+    ) {
+        let v: Vec<A> = a.collect();
+        assert_eq_iters(v.iter(), b);
```

Comment:
could you not just pass ´a`, because it's already an iterator

Reply:
I did try that, however, `a` is an iterator producing (owned) `A`'s, not `&A`'s. Is there a way to still make this work?

Reply:
with `Borrow` you can accept `A` and `&A`

```rust
fn assert_eq_iter<A: std::fmt::Debug + PartialEq + AbsDiffEq + Clone, B: Borrow<A>>(
      a: impl Iterator<Item = B>, 
      b: &[A],
) { 
    assert_eq_iters(a, b);                                                                                                                                                             
}  
fn assert_eq_iters<'a, A, B>( 
    a: impl IntoIterator<Item = B>, 
    b: impl IntoIterator<Item = &'a A>, 
) where
    A: 'a + std::fmt::Debug + PartialEq + AbsDiffEq,
    B: Borrow<A>,
 {
    let mut a_iter = a.into_iter();
    let mut b_iter = b.into_iter();
    loop {
        match (a_iter.next(), b_iter.next()) {
            (None, None) => break,
            (Some(a_item), Some(b_item)) => {
                abs_diff_eq!(a_item.borrow(), b_item);
            }
            _ => {
                panic!("assert_eq_iters: iterators had different lengths");
            }
        }
    }
}
```

Reply:
Awesome, thanks a lot!